### PR TITLE
Execute test suites in parallel but with only one instance of bitcoind and parity

### LINKED ISSUE #2191: Parallelize e2e tests and reuse code
_Follow-up ticket of https://github.com/comit-network/comit-rs/pull/2106_

## Problem

This PR introduces 1 big file (`bitcoin_ethereum.ts`) containing all blockchain-based e2e tests. 
These tests are run sequentially and not in parallel. 

## Goal
Run (at least some) blockchain-based e2e tests in parallel to speed up the execution.

## Note

A few issues I found already:
* we use 1 account on ethereum (parity-dev account) to fund actors. It is hard to parallelize this because of the nonce-issue
* when starting 2 bitcoin nodes our `bcoin` wallet complains, somehow the wallet connected to both blockchains and received competing chains. A solution to that might be to deactivate autodiscovery in the `Chain` class of bcoin, possibly by selecting random ports?

At this stage, just split in two files. It does not matter too much how it is split, we will re-organise the test properly with an issue yet to be created.

The preferred way is to have 1 shared node per blockchain.

The parity issue could be done by using the dev account wallet of parity node instead of signing the transactions with the TS lib.

### PR DESCRIPTION
**Pull request by thomaseizinger**
Resolves #2191.

### CODE REVIEW

**Comment by D4nte** [comment]
Is it supposed to work ? Because I just ran 2c780ff954d5fd5060e707772a735ea0a313866c locally and I am getting errors:

```
/usr/local/Cellar/node@12/12.16.1/bin/node /usr/local/Cellar/yarn/1.22.4/libexec/bin/yarn.js test
yarn run v1.22.4
$ cargo build --bin cnd && tsc
    Finished dev [unoptimized + debuginfo] target(s) in 0.72s
$ jest --forceExit --bail
 PASS  tests/lightning_routes.ts (26.85s)
(node:30042) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 SIGHUP listeners added to [process]. Use emitter.setMaxListeners() to increase limit
(node:30043) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 SIGHUP listeners added to [process]. Use emitter.setMaxListeners() to increase limit
 PASS  tests/rfc003_schema.ts (34.807s)
 PASS  tests/multiple_peers.ts (7.492s)
 PASS  tests/peers_using_ip.ts (21.351s)
(node:30041) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 SIGHUP listeners added to [process]. Use emitter.setMaxListeners() to increase limit
 FAIL  tests/bitcoin_ethereum.ts
  ● Test suite failed to run

    Transaction with the same hash was already imported.

      at getResult (node_modules/ethers/providers/json-rpc-provider.js:40:21)
      at exports.XMLHttpRequest.request.onreadystatechange (node_modules/ethers/utils/web.js:111:30)
      at exports.XMLHttpRequest.dispatchEvent (node_modules/xmlhttprequest/lib/XMLHttpRequest.js:591:25)
      at setState (node_modules/xmlhttprequest/lib/XMLHttpRequest.js:610:14)
      at IncomingMessage.<anonymous> (node_modules/xmlhttprequest/lib/XMLHttpRequest.js:447:13)

 PASS  tests/sanity.ts (57.204s)
 FAIL  tests/ethereum_bitcoin.ts
  ● Test suite failed to run

    Lock file is already being held

      at node_modules/proper-lockfile/lib/lockfile.js:68:47
      at callback (node_modules/graceful-fs/polyfills.js:295:20)

 FAIL  tests/lnd_sanity.ts
  ● Test suite failed to run

    timed out after 60000ms

      52 |         const id = setTimeout(() => {
      53 |             clearTimeout(id);
    > 54 |             reject(new Error(`timed out after ${ms}ms`));
         |                    ^
      55 |         }, ms);
      56 |     });
      57 | 

      at Timeout.<anonymous> (src/utils.ts:54:20)

Test Suites: 3 failed, 5 passed, 8 total
Tests:       22 passed, 22 total
Snapshots:   0 total
Time:        105.077s
Ran all test suites.
A worker process has failed to exit gracefully and has been force exited. This is likely caused by tests leaking due to improper teardown. Try running with --runInBand --detectOpenHandles to find leaks.
Force exiting Jest: Have you considered using `--detectOpenHandles` to detect async operations that kept running after all tests finished?
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

Process finished with exit code 1
```

File: api_tests/src/ledgers/ethereum.ts [hunk 1]
```diff
@@ -47,6 +112,24 @@ export default class EthereumLedger implements LedgerInstance {
...
+ * of the one that was started.
+ */
+class EthereumInstanceProxy implements EthereumInstance {
+    constructor(public readonly rpcUrl: string) {}
+
+    public async start(): Promise<void> {
+        return Promise.resolve();
+    }
+
+    public async stop(): Promise<void> {
```
[the remark this answers is not in the document]
Comment (thomaseizinger) [reply api_tests/src/ledgers/ethereum.ts]: Actually, I am not ensuring this at all :fearful: 

Don't ask me why this still works :D

File: api_tests/src/environment/cleanup.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { rimrafAsync } from "../utils";
+import { EnvGlobal } from "./prepare";
+
+declare var global: EnvGlobal;
+
+module.exports = async () => {
```
[the remark this answers is not in the document]
Comment (thomaseizinger) [reply api_tests/src/environment/cleanup.ts]: True, thanks for pointing that out. Will fix it!

File: api_tests/src/ledgers/ethereum.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (D4nte) [reply api_tests/src/ledgers/ethereum.ts]: Well actually `bitcoind` is not killed if the test fails.
No idea what happens when the test is successful :)

Can you check locally if for you bitcoind is killed when a test fail?

**Comment by thomaseizinger** [comment]
This is on hold until I figure out, how to cleanly shut down the nodes ...

**Pull request closed without merging**